src/methods/remove.ts

```diff
@@ -1,107 +1,84 @@
-import type { RadixNode } from "../trees/radix/node.js";
-import type { Lyra, PropertiesSchema, ResolveSchema, BooleanIndex } from "../types/index.js";
-import { defaultTokenizerConfig } from "../tokenizer/index.js";
-import { removeDocumentByWord } from "../trees/radix/index.js";
-import { flattenObject, getNested } from "../utils.js";
-import { getNodeByKey as getAVLNodeByKey } from "../trees/avl/index.js";
-import * as ERRORS from "../errors.js";
-import { AVLNode } from "../trees/avl/node.js";
-
-/**
- * Removes a document from a database.
- * @param lyra The database to remove the document from.
- * @param docID The id of the document to remove.
- * @example
- * const isDeleted = await remove(db, 'L1tpqQxc0c2djrSN2a6TJ');
- */
-export async function remove<S extends PropertiesSchema>(lyra: Lyra<S>, docID: string): Promise<boolean> {
-  if (!lyra.components?.tokenizer) {
-    lyra.components = {
-      ...(lyra.components ?? {}),
-      tokenizer: defaultTokenizerConfig(lyra.defaultLanguage),
-    };
+import { runMultipleHook, runSingleHook } from "../components/hooks.js";
+import { trackRemoval } from "../components/sync-blocking-checker.js";
+import { createError } from "../errors.js";
+import { Lyra, OpaqueDocumentStore, OpaqueIndex, Schema } from "../types.js";
+
+export async function remove<S extends Schema, I extends OpaqueIndex, D extends OpaqueDocumentStore>(
+  lyra: Lyra<S, I, D>,
+  id: string,
+  language?: string,
+  skipHooks?: boolean,
+): Promise<void> {
```

Comment:
Maybe we can return either `true` or `false` if the function succeeded? That would make it more coherent with other Lyra methods

Reply:
Agree. Will change this.

Reply:
Done. Will be there in the next push.